server/http_integration_test.go

```diff
@@ -2068,6 +2068,255 @@ func TestHTTPServer_AcceptingForwardedUrl(t *testing.T) {
+	defer ResourceOrigin.Close()
+
+	confPath := path.Join("testdata/integration/endpoint_eval/18_couper.hcl")
+	shutdown, hook := newCouperWithTemplate(confPath, test.New(t), map[string]interface{}{"rsOrigin": ResourceOrigin.URL})
+	defer shutdown()
+
+	type expectation struct {
+		Method   string                 `json:"method"`
+		Protocol string                 `json:"protocol"`
+		Host     string                 `json:"host"`
+		Port     int64                  `json:"port"`
+		Path     string                 `json:"path"`
+		Query    map[string][]string    `json:"query"`
+		Origin   string                 `json:"origin"`
+		Url      string                 `json:"url"`
+		Body     string                 `json:"body"`
+		JsonBody map[string]interface{} `json:"json_body"`
+		FormBody map[string][]string    `json:"form_body"`
+	}
+
+	type testCase struct {
+		name   string
+		relUrl string
+		header http.Header
+		body   io.Reader
+		exp    expectation
+	}
+
+	helper := test.New(t)
+	resourceOrigin, err := url.Parse(ResourceOrigin.URL)
+	helper.Must(err)
+
+	port, _ := strconv.ParseInt(resourceOrigin.Port(), 10, 64)
+
+	for _, tc := range []testCase{
+		{
+			"body",
+			"/body",
+			http.Header{},
+			strings.NewReader(`abcd1234`),
+			expectation{
+				Method:   "POST",
+				Protocol: resourceOrigin.Scheme,
+				Host:     resourceOrigin.Hostname(),
+				Port:     port,
+				Path:     "/resource",
+				Query:    map[string][]string{"foo": []string{"bar"}},
+				Origin:   ResourceOrigin.URL,
+				Url:      ResourceOrigin.URL + "/resource?foo=bar",
+				Body:     "abcd1234",
+				JsonBody: map[string]interface{}{},
+				FormBody: map[string][]string{},
+			},
+		},
+		{
+			"json_body",
+			"/json_body",
+			http.Header{"Content-Type": []string{"application/json"}},
+			strings.NewReader(`{"s":"abcd1234"}`),
+			expectation{
+				Method:   "POST",
+				Protocol: resourceOrigin.Scheme,
+				Host:     resourceOrigin.Hostname(),
+				Port:     port,
+				Path:     "/resource",
+				Query:    map[string][]string{"foo": []string{"bar"}},
+				Origin:   ResourceOrigin.URL,
+				Url:      ResourceOrigin.URL + "/resource?foo=bar",
+				Body:     `{"s":"abcd1234"}`,
+				JsonBody: map[string]interface{}{"s": "abcd1234"},
+				FormBody: map[string][]string{},
+			},
+		},
+		{
+			"form_body",
+			"/form_body",
+			http.Header{"Content-Type": []string{"application/x-www-form-urlencoded"}},
+			strings.NewReader(`s=abcd1234`),
+			expectation{
+				Method:   "POST",
+				Protocol: resourceOrigin.Scheme,
+				Host:     resourceOrigin.Hostname(),
+				Port:     port,
+				Path:     "/resource",
+				Query:    map[string][]string{"foo": []string{"bar"}},
+				Origin:   ResourceOrigin.URL,
+				Url:      ResourceOrigin.URL + "/resource?foo=bar",
+				Body:     `s=abcd1234`,
+				JsonBody: map[string]interface{}{},
+				FormBody: map[string][]string{"s": []string{"abcd1234"}},
+			},
+		},
+	} {
+		t.Run(tc.name, func(subT *testing.T) {
+			helper := test.New(subT)
```

Comment:
helper does already exist in parent scope.
```suggestion
			h := test.New(subT)
```

Reply:
Applies to all `helper` references within `t.Run` scope.